build-tools/src/main/kotlin/org/jetbrains/kotlin/benchmark/KotlinNativeBenchmarkingPlugin.kt

```diff
@@ -0,0 +1,102 @@
+package org.jetbrains.kotlin.benchmark
+
+import groovy.lang.Closure
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.Dependency
+import org.gradle.jvm.tasks.Jar
+import org.gradle.util.ConfigureUtil
+import org.jetbrains.kotlin.*
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import org.jetbrains.kotlin.gradle.plugin.mpp.Executable
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTargetPreset
+import org.jetbrains.kotlin.gradle.plugin.mpp.NativeBuildType
+import org.jetbrains.kotlin.konan.target.HostManager
+import javax.inject.Inject
+import kotlin.reflect.KClass
+
+private val NamedDomainObjectContainer<KotlinSourceSet>.jvmMain
+    get() = maybeCreate("jvmMain")
+
+private val Project.jvmWarmup: Int
+    get() = (property("jvmWarmup") as String).toInt()
+
+private val Project.jvmBenchResults: String
+    get() = property("jvmBenchResults") as String
+
+open class KotlinNativeBenchmarkExtension @Inject constructor(project: Project) : BenchmarkExtension(project) {
+    var jvmSrcDirs: Collection<Any> = emptyList()
+    var mingwSrcDirs: Collection<Any> = emptyList()
+    var posixSrcDirs: Collection<Any> = emptyList()
+
+    fun BenchmarkExtension.BenchmarkDependencies.jvm(notation: Any) = sourceSets.jvmMain.dependencies {
```

Comment:
(minor) This method will be available only in Kotlin DSL and only inside the benchmark extension. I.e. this is possible:
```kotlin
benchmark {
    dependencies.jvm(...)
}
```
while this is not:

```kotlin
benchmark.dependencies.jvm(...)
```

But since this plugin is used only by us, this is ok.